src/AppBundle/Form/Type/TagsInputType.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace AppBundle\Form\Type;
+
+use AppBundle\Entity\Tag;
+use AppBundle\Form\DataTransformer\TagArrayToStringTransformer;
+use Doctrine\Common\Persistence\ObjectManager;
+use Symfony\Bridge\Doctrine\Form\DataTransformer\CollectionToArrayTransformer;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Defines the custom form field type used to manipulate tags values across
+ * Bootstrap-tagsinput javascript plugin.
+ *
+ * See http://symfony.com/doc/current/cookbook/form/create_custom_field_type.html
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+class TagsInputType extends AbstractType
+{
+    private $manager;
+
+    public function __construct(ObjectManager $manager)
+    {
+        $this->manager = $manager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->addModelTransformer(new CollectionToArrayTransformer())
+            ->addViewTransformer(new TagArrayToStringTransformer($this->manager))
```

Comment:
this should be a model transformer, not a view transformer. The TextType expects its normalized data to be a string. Changing it to be an array of tags could lead to issues (and such issues could appear in a future version of Symfony if some changes are made, even if there is no such issue today).

Changing the type of the normalized data when extending a type is generally not a supported usage of the type.

the code should look like this:

```php
$builder
    ->addViewTransformer(new TagArrayToStringTransformer($this->manager))
    ->addModelTransformer(new CollectionToArrayTransformer())
;
```